packages/core/src/common/keybinding.ts

```diff
@@ -203,4 +312,42 @@ export class KeybindingRegistry {
         }
     }
 
+    setKeymap(rawKeyBindings: RawKeybinding[]) {
```

Comment:
Couldn't we just create a new keymapsIndex using the given rawKeyBindings?
The check for existing commands and context should just filter down the given `rawKeyBindings` and log errors.
Afterwards we should just do something like
```
   this.keymapsIndex = new KeybindingIndex(this.defaultKeybindingIndex, filteredKeyBindings)
```